android/src/main/kotlin/co/doneservices/callkeep/CallKeepPlugin.kt

```diff
@@ -1,54 +1,246 @@
+                    data.from = "notification"
+                    //send BroadcastReceiver
+                    context?.sendBroadcast(
+                        CallKeepBroadcastReceiver.getIntentIncoming(
+                            requireNotNull(context),
+                            data.toBundle()
+                        )
+                    )
+                    result.success("OK")
+                }
+                "showMissCallNotification" -> {
+                    val data = Data(call.arguments()?: HashMap<String, Any?>())
+                    data.from = "notification"
+                    callKeepNotificationManager?.showMissCallNotification(data.toBundle())
+                    result.success("OK")
+                }
+                "startCall" -> {
+                    val data = Data(call.arguments()?: HashMap<String, Any?>())
+                    context?.sendBroadcast(
+                        CallKeepBroadcastReceiver.getIntentStart(
+                            requireNotNull(context),
+                            data.toBundle()
+                        )
+                    )
+                    result.success("OK")
+                }
+                "endCall" -> {
+                    val data = Data(call.arguments()?: HashMap<String, Any?>())
+                    context?.sendBroadcast(
+                        CallKeepBroadcastReceiver.getIntentEnded(
+                            requireNotNull(context),
+                            data.toBundle()
+                        )
+                    )
+                    result.success("OK")
+                }
+                "endAllCalls" -> {
+                    val calls = getDataActiveCalls(context)
+                    calls.forEach {
+                        if(it.isAccepted) {
+                            context?.sendBroadcast(
+                                CallKeepBroadcastReceiver.getIntentEnded(
+                                    requireNotNull(context),
+                                    it.toBundle()
+                                )
+                            )
+                        }else {
+                            context?.sendBroadcast(
+                                CallKeepBroadcastReceiver.getIntentDecline(
+                                    requireNotNull(context),
+                                    it.toBundle()
+                                )
+                            )
+                        }
+                    }
+                    removeAllCalls(context)
+                    result.success("OK")
+                }
+                "activeCalls" -> {
+                    result.success(getDataActiveCallsForFlutter(context))
+                }
+                "getDevicePushTokenVoIP" -> {
+                    result.success("")
+                }
+            }
+        } catch (error: Exception) {
+            result.error("error", error.message, "")
+        }
+    }
```

Comment:
It would be great to use https://pub.dev/packages/pigeon to type this.

Reply:
Agreed but let's leave it for another PR?